vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLQueryTest.java

```diff
@@ -268,6 +274,116 @@ public void testPreparedCollector(TestContext ctx) {
     }));
   }
 
+  private static class CollectorBase implements Collector<Row, Object, Object> {
+    @Override
+    public Supplier<Object> supplier() {
+      return () -> null;
+    }
+
+    @Override
+    public BiConsumer<Object, Row> accumulator() {
+      return (a, t) -> {
+
+      };
+    }
+
+    @Override
+    public BinaryOperator<Object> combiner() {
+      return (a, a2) -> null;
+    }
+
+    @Override
+    public Function<Object, Object> finisher() {
+      return a -> null;
+    }
+
+    @Override
+    public Set<Characteristics> characteristics() {
+      return Collections.emptySet();
+    }
+  }
+
+  @Test
+  public void testCollectorFailureProvidingSupplier(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public Supplier<Object> supplier() {
+        throw cause;
+      }
+    });
+  }
+
+  @Test
+  public void testCollectorFailureInSupplier(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public Supplier<Object> supplier() {
+        return () -> {
+          throw cause;
+        };
+      }
+    });
+  }
+
+  @Test
+  public void testCollectorFailureProvidingAccumulator(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public BiConsumer<Object, Row> accumulator() {
+        throw cause;
+      }
+    });
+  }
+
+  @Test
+  public void testCollectorFailureInAccumulator(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public BiConsumer<Object, Row> accumulator() {
+        return (o, row) -> {
+          throw cause;
+        };
+      }
+    });
+  }
+
+  @Test
+  public void testCollectorFailureProvidingFinisher(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public Function<Object, Object> finisher() {
+        throw cause;
+      }
+    });
+  }
+
+  @Test
+  public void testCollectorFailureInFinisher(TestContext ctx) {
+    RuntimeException cause = new RuntimeException();
+    testCollectorFailure(ctx, cause, new CollectorBase() {
+      @Override
+      public Function<Object, Object> finisher() {
+        return o -> {
+          throw cause;
+        };
+      }
+    });
+  }
+
+  private void testCollectorFailure(TestContext ctx, Throwable cause, Collector<Row, Object, Object> collector) {
+    MySQLConnection.connect(vertx, options, ctx.asyncAssertSuccess(conn -> {
+      conn.query("SELECT * FROM collectorTest WHERE id = 1", collector, ctx.asyncAssertFailure(result -> {
+        ctx.assertEquals(cause, result);
+        conn.close();
+      }));
+    }));
+  }
+
```

Comment:
Maybe we can move these new tests and the `Collector` related tests to the `vertx-sql-client` later.

Reply:
yes it would be good thing to do